File: Easy/Binary Search/binarysearch.java
class Solution {
    public int search(int[] nums, int target) {
        int low = 0, mid = 0, high = nums.length - 1;
        while (low <= high){
            mid = (low + high)/2;
            if (nums[mid] < target) low = mid + 1;
            else if (nums[mid] > target) high = mid - 1;
            else return mid; // element found
        }
        return -1; //element not found
    }
}

File: Easy/Binary Search/binarysearch.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        low,mid,high = 0,0,len(nums)-1
        while low <= high:
            mid = (high + low) // 2
            # If target is greater, ignore left half
            if nums[mid] < target:
                low = mid + 1
            # If target is smaller, ignore right half
            elif nums[mid] > target:
                high = mid - 1
            # target is present at mid
            else:
                return mid
        # target was not present
        return -1

